libnavigation-util/src/test/java/com/mapbox/navigation/utils/timer/MapboxTimerTest.kt

```diff
@@ -0,0 +1,42 @@
+package com.mapbox.navigation.utils.timer
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+
+class MapboxTimerTest {
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(Dispatchers.Default)
+    }
+
+    @Test
+    fun start() = runBlocking {
+        var lambdaCalled = false
+        val testLambda = {  lambdaCalled = true }
+
+        MapboxTimer(0L, testLambda).start()
+        delay(10)
+
+        assertTrue(lambdaCalled)
+    }
+
+    @Test
+    fun stop() = runBlocking {
+        var counter = 0
+        val testLambda = { counter += 1 }
+
+        val timer = MapboxTimer(100L, testLambda)
+        timer.start()
+        delay(150)
+        timer.stop()
+        delay(150)
```

Comment:
why do you need one more delay?

Reply:
I"m testing the stop method in this test. If the last delay is removed and stop() is commented out the test will pass erroneously.  The delay is needed so that the implementation will tick one more time and the implementation's check of isActive should indicate that the timer is not active and therefore the lambda wouldn't get called an additional time.